### ISSUE TITLE
AI cannot buy unbuildable constructions

### ISSUE THREAD

**Issue by SeventhM** [yairm210/Unciv#10581]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Game Version

4.9.2

### Describe the bug

Reporting a bug someone on discord noticed:

AIs cannot buy units or buildings that are purchasable via a stat if it has the unbuildable unique. This is because the AI cannot purchase objects unless it is currently building it. See [here](https://github.com/yairm210/Unciv/blob/5396c41efc6743d6da06de33ed81a457b856cd64/core/src/com/unciv/logic/automation/civilization/UseGoldAutomation.kt#L25). Funny side effect of how the function is written is it can only ever purchase one object a turn per city

Seems like the AI purchase logic in general needs fixing up

### Steps to Reproduce

N/A

### Expected Behavior

I will pretend the answer is "N/A"

### Operating System

Android

**Comment by SeventhM** [comment]
> we know you're a proficient coder

Yeah, but I'm not much of an AI dev. I'd probably copy over some of the build construction logic and call it a day. I know Tuvus has been making adjustments to the AI, so this issue is more making note of something to fix than just a setup for me to fix

**Comment by SeventhM** [comment]
Removing Stale label, but only as an aide to #11264

**Comment by tuvus** [comment]
I am unstaling this since it is important and is relatively high on my list. I'll see when I can get to it.
